sklearn/decomposition/_base.py

```diff
@@ -60,7 +60,7 @@ def get_precision(self):
         # handle corner cases first
         if self.n_components_ == 0:
             return np.eye(n_features) / self.noise_variance_
-        if self.n_components_ == n_features:
+        if self.n_components_ == min(self.explained_variance_.shape):
```

Comment:
`self.explain_variance_` is always of shape `n_components` so I'm not sure this is the intended fix? I think that condition would always be True